### PR TITLE
#20, #23 feat: redirect after auth and logout button

### PR DESCRIPTION
**Pull request by SergioAJS**

### CODE REVIEW

File: mw-webapp/src/component/usersBlock/UsersBlock.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/component/usersBlock/UsersBlock.tsx:20]: Better create const instead of Object.values....

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:7]: Is it possible to rename this consts or it's naming from firebase? For make it more similar - for example to change handleGoogleSignIn -> handleLogIn?

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:23]: I think better to move out the function and then just call it inside useEffect

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:48]: Again if possible bette rmove out function. And call it inside useEffect. Also no eed to use 2 useEffect. You can call few functions in 1 useEffect

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:55]: Need to use component button, not new buttons. I think that task #23 will be ready after that.

Comment (Ekaterina1994) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:55]: Need to use component button, not new buttons. I think that task #23 will be ready after that.

File: mw-webapp/src/utils/useAuth.ts
Comment (Ekaterina1994) [inline mw-webapp/src/utils/useAuth.ts:9]: Better create an interface UserDataProps with these properties

File: mw-webapp/src/utils/useAuth.ts
Comment (Ekaterina1994) [inline mw-webapp/src/utils/useAuth.ts:43]: Also maybe better moveout these functions to keep single responsibility

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx:7]: Done

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx:55]: Done

Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx:55]: Done

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx]: Done

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx:20]: Done

File: mw-webapp/src/utils/useAuth.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/useAuth.ts]: Done

File: mw-webapp/src/utils/useAuth.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/useAuth.ts]: After that, the existence of the useAuth hook became senseless and I deleted it.
Done

File: mw-webapp/src/component/usersBlock/UsersBlock.tsx
Comment (tritonJS826) [inline mw-webapp/src/component/usersBlock/UsersBlock.tsx:20]: it is not a converter. It is `usersList`

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (tritonJS826) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:17]: why does getNewUserCredentials returns void, and its interface include navigate and part of url as a string? Fix all this please)

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (tritonJS826) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:16]: this callback wil be invoked after unmounting  Auth page. Why do you need to handle auth state in this case and what does it mean. Use more approproate abstraction or logic.

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (tritonJS826) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:23]: dont use ternarian operator in multiline expressions if possibe. Maybe in this case it will be more appropriate to render the same buttom with different props depeds on does user exist

File: mw-webapp/src/utils/auth/getNewUserCredentials.ts
Comment (tritonJS826) [inline mw-webapp/src/utils/auth/getNewUserCredentials.ts:6]: this function violates single responsibility and it does not get anything. Refactor required

File: mw-webapp/src/utils/auth/getNewUserCredentials.ts
Comment (tritonJS826) [inline mw-webapp/src/utils/auth/getNewUserCredentials.ts:22]: why can't you use something simplier like `??` operator or add helper?

File: mw-webapp/src/utils/auth/handleAuthState.ts
Comment (tritonJS826) [inline mw-webapp/src/utils/auth/handleAuthState.ts:10]: the same as setUserFunction(currentUser ?? null),

File: mw-webapp/src/utils/auth/handleLogIn.ts
Comment (tritonJS826) [inline mw-webapp/src/utils/auth/handleLogIn.ts:12]: move to helper or use simplier method like '??' operator

File: mw-webapp/src/component/usersBlock/UsersBlock.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/component/usersBlock/UsersBlock.tsx:22]: Please think about naming. Inside function renderUsers we work with users so I think better use const users for map it. You can change naming for state const

File: mw-webapp/src/utils/auth/handleLogOut.ts
Comment (tritonJS826) [inline mw-webapp/src/utils/auth/handleLogOut.ts:12]: helper or simplier

File: mw-webapp/src/component/usersBlock/UsersBlock.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/component/usersBlock/UsersBlock.tsx:22]: Please think about naming. Inside function renderUsers we work with users so I think better use const users for map it. You can change naming for state const

File: mw-webapp/src/utils/auth/handleLogOut.ts
Comment (tritonJS826) [inline mw-webapp/src/utils/auth/handleLogOut.ts:6]: what heppend for user if sign out throw error? Nothing except alert? Is it a good scenario?

File: mw-webapp/src/utils/auth/writeNewUserData.ts
Comment (tritonJS826) [inline mw-webapp/src/utils/auth/writeNewUserData.ts:11]: it is very hard to managed widespreaded strings in enterprise projects. move to service

**Review by tritonJS826 — CHANGES_REQUESTED** [review]

**Comment by tritonJS826** [comment]
Call me in the telegram when you will solve all issues and will need my approve (after Kates approve, please)

File: mw-webapp/src/utils/auth/writeNewUserData.ts
Comment (Ekaterina1994) [inline mw-webapp/src/utils/auth/writeNewUserData.ts:9]: Check syntax. Where is ;

File: mw-webapp/src/utils/auth/writeNewUserData.ts
Comment (Ekaterina1994) [inline mw-webapp/src/utils/auth/writeNewUserData.ts:10]: Is it possible don't use destructure? If image that user has 100 properties it will be hard to write it all

Comment (Ekaterina1994) [inline mw-webapp/src/utils/auth/writeNewUserData.ts:10]: maybe this const better rename to createNewUser/ addNewUser? Or something like this? What do you think?

File: mw-webapp/src/utils/auth/handleAuthState.ts
Comment (Ekaterina1994) [inline mw-webapp/src/utils/auth/handleAuthState.ts:4]: Check naming. Function should have the same name as file

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/component/usersBlock/UsersBlock.tsx
Comment (SergioAJS) [reply mw-webapp/src/component/usersBlock/UsersBlock.tsx:22]: Done

File: mw-webapp/src/component/usersBlock/UsersBlock.tsx
Comment (SergioAJS) [reply mw-webapp/src/component/usersBlock/UsersBlock.tsx:20]: Done

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx:23]: Done

File: mw-webapp/src/utils/auth/handleAuthState.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/handleAuthState.ts:10]: Done, thanks

File: mw-webapp/src/utils/auth/handleAuthState.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/handleAuthState.ts:4]: Done

File: mw-webapp/src/utils/auth/handleAuthState.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/handleAuthState.ts:7]: Done

File: mw-webapp/src/utils/auth/writeNewUserData.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/writeNewUserData.ts:10]: Done

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx:17]: This function writes user data to database.
It includes navigate to redirect to the page with a table only if the user logged in. If the user already logged in three is no redirection to the page with the table. I write in such way because we don't have logout button on the table page.
Npw I delete navigate params from this function.

File: mw-webapp/src/utils/auth/getNewUserCredentials.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/getNewUserCredentials.ts:6]: Done

File: mw-webapp/src/utils/auth/writeNewUserData.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/writeNewUserData.ts:11]: I remove destructing if you mean it

File: mw-webapp/src/utils/auth/getNewUserCredentials.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/getNewUserCredentials.ts:22]: Done, if understand you right

File: mw-webapp/src/utils/auth/handleLogIn.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/handleLogIn.ts:12]: Done

File: mw-webapp/src/utils/auth/handleLogOut.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/handleLogOut.ts:12]: Done

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (Ekaterina1994) [reply mw-webapp/src/pages/authPage/AuthPage.tsx]: Victor mean that if you call function get... then this function should return something so if this function doesn't return anything  need to change naming

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:18]: Do you need to use return getNewUserCredentials or maybe just call it will be enough

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:27]: Move out text of buttons to consts

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (Ekaterina1994) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:24]: create consts with paths. Think maybe not in this file

File: mw-webapp/src/utils/auth/getNewUserCredentials.ts
Comment (Ekaterina1994) [inline mw-webapp/src/utils/auth/getNewUserCredentials.ts:5]: change naming . get.. always return values

**Review by Ekaterina1994 — CHANGES_REQUESTED** [review]

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx:18]: Now it's enough. Early this function doesn't write user to DB if I call it in useEffect.
Done

File: mw-webapp/src/utils/auth/getNewUserCredentials.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/getNewUserCredentials.ts:5]: Done

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx]: When we need to use path consts somewhere else we will create separate file.
For now, I create this const in the same file

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx]: Done

**Review by Ekaterina1994 — APPROVED** [review]

File: mw-webapp/src/component/usersBlock/UsersBlock.tsx
Comment (tritonJS826) [inline mw-webapp/src/component/usersBlock/UsersBlock.tsx:20]: what does this line do? you are trying to get array from array with the same values or i miss siimething?

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (tritonJS826) [inline mw-webapp/src/pages/authPage/AuthPage.tsx:13]: we have pages.tsx file with router related data. remember about encapsulation.

File: mw-webapp/src/utils/auth/handleLogIn.ts
Comment (tritonJS826) [inline mw-webapp/src/utils/auth/handleLogIn.ts:6]: should it be in service layer? what heppend for user sign in throw error? Nothing except alert? Is it a good scenario?

File: mw-webapp/src/utils/auth/handleUserCredentials.ts
Comment (tritonJS826) [inline mw-webapp/src/utils/auth/handleUserCredentials.ts:7]: i dont understand why network related methods are in the utils

Comment (tritonJS826) [reply mw-webapp/src/utils/auth/handleUserCredentials.ts:7]: @Ekaterina1994

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/utils/auth/handleLogOut.ts
Comment (Ekaterina1994) [reply mw-webapp/src/utils/auth/handleLogOut.ts:6]: Не менял поведение при появлении ошибки при Logout, потому что не понял, что еще надо сделать? Редирект на страницу авторизации?
Не уверен, что правильно написал обработчик ошибок с использованием ??

File: mw-webapp/src/component/usersBlock/UsersBlock.tsx
Comment (Ekaterina1994) [reply mw-webapp/src/component/usersBlock/UsersBlock.tsx:20]: It''s because of on fb collections is objects with key type string. And when we mapping this data we need firstly convert it to array where key will be 0,1,2... If you know how to map objects with any keys tell us

Comment (SergioAJS) [reply mw-webapp/src/component/usersBlock/UsersBlock.tsx:20]: If the keys in the database are sequential - 0, 1, 2...n, then we get an array at once and there are no errors, otherwise - an object, and on the object map does not work.
At first, I thought that the error was due to the fact that the key I use as userId contains letters. So I tried first removing all letters and leaving only numbers before using userId as a key. But that didn't work because the resulting number was large and still returned an object
Then I tried to get the length of the users collection from the database to assign the key based on the number of users. This didn't work because I couldn't get the length of the collection before writing a new user to the database
In the end, I decided to use Object.values to get an array of user data.

File: mw-webapp/src/utils/auth/handleLogIn.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/handleLogIn.ts:6]: Didn't change the behavior when Logout error appears because I didn't understand what else needs to be done? Redirect to the authorization page?

File: mw-webapp/src/utils/auth/handleUserCredentials.ts
Comment (SergioAJS) [reply mw-webapp/src/utils/auth/handleUserCredentials.ts:7]: I will move it to the appropriate folder if you tell me where it should be

File: mw-webapp/src/pages/authPage/AuthPage.tsx
Comment (SergioAJS) [reply mw-webapp/src/pages/authPage/AuthPage.tsx:13]: Use path from pages.tsx file
By the way, the paths there are specified as strings, not as constants

**Pull request merged**
